### ISSUE TITLE
per site option to hide specified element(s)

### ISSUE THREAD

**Issue by jadedgnome** [gildas-lormeau/SingleFile#1635]
I often use the SingleFile extension to save webpages, but sometimes I need to hide or remove certain elements (such as sensitive data, advertisements, or other nuisance elements) before saving. Currently, I need to rely on another extension to delete or hide these elements, which can be a bit cumbersome. It would be fantastic if SingleFile could include a per-site option that allows users to specify elements to hide or remove before saving the page.

Proposed Feature:

    A setting in SingleFile where users can define specific elements (using CSS selectors) to be hidden or removed before the page is saved.
    This feature could be configured per site to allow different settings for different websites.
    A simple UI to add/remove/edit the selectors for each site, or an option to use the browser’s built-in developer tools to pick the elements directly.

Use Case:
This feature would allow users to:

    Hide or remove sensitive information (such as personal data or login details).
    Avoid saving unnecessary elements like pop-ups, banners, or ads.
    Have a more seamless experience without needing to use an additional add-on or manually remove elements.

Additional Considerations:

    It would be useful if the configuration could be as simple as a list of CSS selectors for each site that need to be hidden or deleted.
    An option to reset the configuration for a site or disable the feature entirely.

**Comment by gildas-lormeau** [comment]
I agree it could be useful. Out of curiosity, have you tried using a user-script to do this? See https://github.com/gildas-lormeau/SingleFile/wiki/How-to-execute-a-user-script-before-a-page-is-saved

**Comment by jadedgnome** [comment]
i didn't know about 😮 , i'll try to see if i can get it to work for me. thanks!

**Comment by JokerQyou** [comment]
I'd like this as well, though for a different usage. Many browser extensions inject extra elements into page content, for example OpenAI translator would inject a popup container into the page, and it used negative z-index to hide itself (rather than using `display: none`), so SingleFile would save this element into final html file. This isn't much a big deal, but it does leak some metadata (about what extensions the original user installed).

**Comment by gildas-lormeau** [comment]
A new option "remove specific elements" will allow you to provide CSS selectors of elements to remove from the page in the next version.
